src/NuGet.Core/NuGet.Build.Tasks/RestoreTask.cs

```diff
@@ -242,5 +314,243 @@ public void Dispose()
         {
             _cts.Dispose();
         }
+
+#if IS_DESKTOP
+        private async Task<RestoreSummary> PerformNuGetV2RestoreAsync(Common.ILogger log, DependencyGraphSpec dgFile)
+        {
+            string globalPackageFolder = null;
+            string repositoryPath = null;
+            string firstPackagesConfigPath = null;
+            IList<PackageSource> packageSources = null;
+
+            var installedPackageReferences = new HashSet<Packaging.PackageReference>(new PackageReferenceComparer());
+
+            ISettings settings = null;
+
+            foreach (PackageSpec packageSpec in dgFile.Projects.Where(i => i.RestoreMetadata.ProjectStyle == ProjectStyle.PackagesConfig))
+            {
+                globalPackageFolder = globalPackageFolder ?? packageSpec.RestoreMetadata.PackagesPath;
+                repositoryPath = repositoryPath ?? packageSpec.RestoreMetadata.RepositoryPath;
+
+                if (packageSources == null)
+                {
+                    packageSources = new List<PackageSource>();
+                    if (!RestoreNoCache)
+                    {
+                        if (!string.IsNullOrEmpty(globalPackageFolder) && Directory.Exists(globalPackageFolder))
+                        {
+                            packageSources.Add(new FeedTypePackageSource(globalPackageFolder, FeedType.FileSystemV3));
+                        }
+                    }
+
+                    packageSources.AddRange(packageSpec.RestoreMetadata.Sources);
+                }
+
+                settings = settings ?? Settings.LoadSettingsGivenConfigPaths(packageSpec.RestoreMetadata.ConfigFilePaths);
+
+                var packagesConfigPath = Path.Combine(Path.GetDirectoryName(packageSpec.RestoreMetadata.ProjectPath), NuGetConstants.PackageReferenceFile);
+
+                firstPackagesConfigPath = firstPackagesConfigPath ?? packagesConfigPath;
+
+                installedPackageReferences.AddRange(GetInstalledPackageReferences(packagesConfigPath, allowDuplicatePackageIds: true));
+            }
+
+            PackageSourceProvider packageSourceProvider = new PackageSourceProvider(settings);
+            var sourceRepositoryProvider = new CommandLineSourceRepositoryProvider(packageSourceProvider);
+            var nuGetPackageManager = new NuGetPackageManager(sourceRepositoryProvider, settings, repositoryPath);
+
+            // TODO: different default?  Allow user to specify?
+            var packageSaveMode = Packaging.PackageSaveMode.Defaultv2;
```

Comment:
You can't specify a `PackageSaveMode` when using the `RestoreTask`, should I add a parameter?  And what should the default be?